src/app/community/community.component.ts

```diff
@@ -84,11 +66,38 @@ export class CommunityComponent implements OnInit, AfterViewInit {
     // With object destructuring colon means different variable name assigned, i.e. 'id' rather than '_id'
       const { _id: id, _rev: rev } = community;
       community.registrationRequest = change;
-      this.couchService.put('communityregistrationrequests/' + id + '?rev=' + rev, community)
-        .subscribe((data) => {
-          this.updateRev(data, this.communities.data);
-          this.editDialog.close();
-        }, (error) => this.editDialog.componentInstance.message = 'There was a problem accepting this community');
+      if (change === 'delete' || change === 'reject' || change === 'unlink') {
+        this.couchService.put('communityregistrationrequests/' + id + '?rev=' + rev, community)
+          .subscribe((data) => {
+            this.updateRev(data, this.communities.data);
+            this.editDialog.close();
+          }, (error) => this.editDialog.componentInstance.message = 'There was a problem accepting this community');
+      }
+      if (change === 'accept') {
+        community.registrationRequest = 'accepted';
+        this.couchService.get('_users/_all_docs?include_docs=true')
+          .subscribe((data) => {
+            data.rows.map(data => {
```

Comment:
This block includes 4 levels of nested CouchDB requests/subscribes, 3 of which are nested in a `map`.

1. Please **do not** loop over even 1 couch request.  Use [_bulk_docs](http://docs.couchdb.org/en/2.1.1/api/database/bulk-api.html#db-bulk-docs) instead.
2. Do not nest subscribes.  Create different functions and/or use the [switchMap](http://reactivex.io/rxjs/class/es6/Observable.js~Observable.html#instance-method-switchMap) rxjs operator to clean this up.